packages/ui/src/components/pages/Accounts/AccountControls/RenameAccountDialog.tsx

```diff
@@ -0,0 +1,79 @@
+import { ChangeEvent, FormEvent, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { toast } from 'react-toastify';
+import { useEffectOnce } from 'react-use';
+import { Button, Dialog, DialogContent, DialogContentText, TextField } from '@mui/material';
+import DialogTitle from 'components/shared/DialogTitle';
+import EmptySpace from 'components/shared/misc/EmptySpace';
+import useDialog from 'hooks/useDialog';
+import { useWalletState } from 'providers/WalletStateProvider';
+import { AccountInfoExt, Props } from 'types';
+import { EventName, EventRegistry } from 'utils/eventemitter';
+
+export default function RenameAccountDialog({}: Props): JSX.Element {
+  const { keyring } = useWalletState();
+  const { t } = useTranslation();
+  const { open, doOpen, doClose } = useDialog();
+  const [name, setName] = useState<string>('');
+  const [account, setAccount] = useState<AccountInfoExt>();
+
+  const onOpen = (account: AccountInfoExt) => {
+    setAccount(account);
+    setName(account.name!);
+    doOpen();
+  };
+
+  useEffectOnce(() => {
+    EventRegistry.on(EventName.OpenRenameAccountDialog, onOpen);
+    return () => {
+      EventRegistry.off(EventName.OpenRemoveAccountDialog, onOpen);
+    };
+  });
+
+  const onClose = () => {
+    doClose(() => setAccount(undefined));
+  };
+
+  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setName(event.currentTarget.value);
+  };
+
+  const doRename = async (event: FormEvent) => {
+    event.preventDefault();
+    try {
+      await keyring.renameAccount(account!.address, name);
+      onClose();
+      toast.success(t<string>('Account renamed'));
+    } catch (e: any) {
+      toast.error(t<string>(e.message));
+    }
+  };
+
```

Comment:
Similar to the `RemoveAccountDialog`, we should add an early return if account if empty:
```jsx
if (!account) return <></>;
```